lib/mocha/expectation.rb

```diff
@@ -566,9 +566,8 @@ def satisfied?
     end
 
     # @private
-    def invoke(*arguments)
+    def invoke(invocation = Invocation.new(@mock, @method_matcher.mocha_inspect))
```

Comment:
It feels a bit ugly to reference instance variables from an argument default value. Have you done this to avoid breaking a load of tests? If so, it would be helpful to explain that in the commit note.

Reply:
I've opened #414 to capture this.